corehq/apps/hqmedia/tasks.py

```diff
@@ -190,24 +190,24 @@ def _zip_ccs_files(fpath, files, current_progress, file_progress, file_count, co
                 file_compression = zipfile.ZIP_STORED if extension in MULTIMEDIA_EXTENSIONS else compression
                 z.writestr(path, data, file_compression)
                 current_progress += file_progress / file_count
-                if track_progress:
-                    DownloadBase.set_progress(build_application_zip, current_progress, 100)
+                if task:
```

Comment:
@esoergel yes simon pointed that earlier.
should not we avoid few method calls if we can? So i already know that if task is not present, why set progress?

Reply:
Method calls have so little overhead in Python that it's not worth factoring in as a performance consideration.  I'd make the decision entirely based on what makes the cleanest code.  I think it's cleaner to avoid checking whether `task` exists every time it's referenced, considering that `task=None` is an accepted part of the API.  You're certainly free to disagree with that opinion, however.

Reply:
okay, updated.